### DISCUSSION TITLE
Prevent reserved variables to be passed on to the view

### DISCUSSION THREAD

**Discussion by LordSimal** [laravel/framework#50225] [Ideas]
Thought this was a [livewire issue](https://github.com/livewire/livewire/discussions/8003) but apperantly this could be improved in laravel itself.

I was trying to output validation errors for a variable which worked without any problems in another component.

But was rather confused when I got the error:
```
Call to a member function getBag() on array
```
when doing a simple
```
@error('editingCategoryName')
    <x-input-error :messages="$message" class="mt-2"/>
@enderror
```

Reason behind that is the fact, that inside my `render()` method I did this:
```
$categories = Category::query()->with(['bankEntries'])
    ->orderBy('lft')
    ->get()
    ->toTree();
$hasErrors = Category::isBroken();
$treeErrors = $hasErrors ? Category::countErrors() : [];

return view('livewire.category.index', [
    'categories' => $categories,
    'hasErrors' => $hasErrors,
    'errors' => $treeErrors,
]);
```

So I unintentionally overwrote the `$errors` variable in the view which is present by default.

My proposed solution would be to throw an exception if variables are passed on to the `view()` method if they contain keys which are known, reservered variables for default functionality.

**Comment by AlliBalliBaba** [comment]
I like the idea of a warning if reserved variables are overwritten. Throwing an exception would probably break too many existing projects though where people don't intend to use the reserved variable. 

Mavbe the exception handler for views could add a notice to the actual exception message, don't know how feasible that would be though.
